crates/valence_protocol/src/item.rs

```diff
@@ -1,12 +1,13 @@
 use std::io::Write;
 
+use serde::{Deserialize, Serialize};
 pub use valence_generated::item::ItemKind;
 use valence_nbt::Compound;
 
 use crate::{Decode, Encode};
 
 /// A stack of items in an inventory.
-#[derive(Clone, PartialEq, Debug, Default)]
+#[derive(Serialize, Deserialize, Clone, PartialEq, Debug, Default)]
```

Comment:
Should we put these behind a `serde` feature?

Reply:
I think serializing Itemstacks isnt that rare of a case.
Im currently creating a minigame server and i had to save items to a json file (for a ingame shop menu).

Im fine with a either.

Reply:
Ok, then let's forego the serde feature for now.